\documentclass{bioinfo}
\copyrightyear{2016} \pubyear{2016}

\usepackage{gensymb}
\usepackage{tikz}
\usetikzlibrary{calc}

\usepackage{epstopdf}


\access{Advance Access Publication Date: 23 Oct 2016}
\appnotes{Manuscript Category}

\begin{document}
\firstpage{1}

\subtitle{Computational and Systems Biology}

\title[Model inference from protein time-courses in Hematopoietic Stem Cells (HSC)]{Model inference from protein time-course in Hematopoietic Stem Cells (HSC)}
\author[Pandu Raharja, Rene Schoeffel, Michael Strasser and Carsten Marr]{Pandu Raharja\,$^{\text{\sfb 1,}\text{\sfb 2,}\text{\sfb 3,}*}$, Rene Schoeffel\,$^{\text{\sfb 1,}\text{\sfb 2,}\text{\sfb 3}}$, Michael Strasser\,$^{\text{\sfb 3}}$ and Carsten Marr\,$^{\text{\sfb 3,}*}$}
\address{$^{\text{\sf 1}}$Technische Universit\"at M\"unchen, Fakult\"at f\"ur Informatik, Boltzmannstr. 3, 85748 Garching bei M\"unchen, Germany\\
$^{\text{\sf 2}}$Ludwig-Maximilians-Universit\"at M\"unchen, Professor-Huber-Platz 2, 80539 M\"unchen, Germany\\
$^{\text{\sf 3}}$Helmholtz Zentrum M\"unchen, Institute of Computational Biology, Ingolst\"adter Landstr. 1
85764 Neuherberg, Germany.}

\corresp{$^\ast$To whom correspondence should be addressed.}

\history{Received on 23/09/2016; revised on 14/10/2016; accepted on 21/10/2016}

\editor{Associate Editor: Jan Quell}

\abstract{\textbf{Motivation:} 
In single cell setting, the dynamics of the systems governing the cell -- gene expression being one -- are heavily influenced by the stochasticity of each involving element. These stochastic dynamics in gene expression appears to convey more information about the underlying mechanism of gene expression processes than it would be otherwise assumed through the study of population average.\\
\textbf{Results:}In this paper we presented a particle filtering-based algorithm and a framework that is capable of inferring parameters underlying the stochastic models from single cell expression data. This framework was then applied on time-lapsed microscopy data of two transcription factors (\texttt{Pu.1} and \texttt{Gata.1}) in murine blood stem cells. It is thought that both transcription factors play decisive roles in several stages of stem cell differentiation, especially the differentiation of myeloid progenitors. Our results provide several insights into the dynamics of blood stem cells maturation and specifically in the single cell environment. We managed to gain several valuable insights on the interaction dynamics between two transcription factors and the outcome of cell maturation. While doing so, we managed to developed highly flexible general purpose highly parallelized framework to model dynamical systems using particle filtering. The general framework software is now available as free software for anyone to use.\\
\textbf{Availability:} The data are available upon request. A general framework for determining the parameters that best explain the data were published under open source license. The complete source code of the program is publicly accessible at \href{https://github.com/raharjaliu/PFInfer}{https://github.com/raharjaliu/PFInfer}.\\
\textbf{Contact:} \href{pandu.raharja@tum.de}{pandu.raharja@tum.de}\\
\textbf{Supplementary information:} Supplementary data are available at \textit{Bioinformatics}
online.}

\maketitle

\section{Introduction}

The advances in gene expression measurement techniques, coupled with equally rapid advances in single cell analytics, have allowed the more recent high resolution single cell analysis of cell dynamics \citep{Feigelman16, Hoppe16}. This has enabled us, for example, to look deeper into the dynamics of stem cell maturation. In recent years, several genes that potentially are involved in the decision making mechanism of cell maturation going from Inner Mass Cell (IMC) all the way to somatic cells have been discovered and widely known by now \citep{Graf09}. While this has enable us to study cell processes more thoroughly, the deterministic population-averaged nature of cell dynamics was becoming less and less apparent while the stochastic nature of single cell dynamics was becoming dominating as the analysis went down onto single cell resolution \citep{Elowitz02}. Hence our motivation to apply inference methods that are capable to infer dynamics of the systems that are heavily influenced by non-deterministic factors.

For this project we use particle filtering to learn more about aforementioned dynamics. We apply this technique on our time-dependent expression data of single cells containing concentration of both transcription factors. The expression data were measured from time-lapsed microscopy of fluorescent-tagged proteins' expression level of the cells. The approach of the experiment would be further described in the next section followed by the theoretical aspects of the particle filtering and other methods used in this project.

In this paper we focus on the maturation lines of hematopoietic stem cells \citep{Orkin08}. The cell lines start with Long Term Hematopoietic Stem Cells (LT-HSC), which is the progenitor of all blood cells. During its life, LT-HSC could either divide into two LT-HSC or turn into Short Term Hematopoietic Stem Cell (ST-HSC). SH-HSC will then undergo a transition process into either Common Myeloid Progenitor (CMP) or Common Lymphoid Progenitor (CLP). As the names suggest, both cells are the ancestor of all myeloid cells and lymphocites, respectively. Each would then further maturate into their respective intermediate cells and eventually the mature somatic cells (Red Blood Cells, Megakaryocytes, Mast Cells, Eosinophils, Neutrophils and Macrophages from CMP and B and T Lymphocytes from CLP; see Figure \ref{fig:01}).

The decision process of CMP transitioning into either MEP and GMP is the focus of our project. Two transcription factors, \texttt{Pu.1}and \texttt{Gata.1} are thought to influence the decision process \citep{Graf09}. Moreover, both transcription factors are known to inhibit each other. It is thought that the whole cross inhibition dynamics between the two has an impact on the fate of Common Myeloid Progenitor. This dual-agent dynamics is thought to behave in a way that is self-fulfilling. That is, a slight change in concentration of transcription factors favoring one state over another will have stupendous impact on cell fate decision \citep{Zhang03}.

\begin{figure}[h]
\includegraphics[width=8cm]{figures/homatopoietic_focus}
\caption{Maturation cascade of intermediate Common Myeloid Progenitor (CMP) to Megakaryocyte-Eerythroid Progenitor (MEP) and Granulocyte/Macrophage Progenitor (GMP). MEP would then maturate into somatic Red Blood Cells and Megakaryocytes while GMP would undergo maturation into Granulocytes and Monocyptes. For each process there are known factors which are known to interact antagonistically on maturation decision. Our CMP to MEP and GMP decision stands on the top of the figure with two transcription factors of interest, \texttt{Pu.1} and \texttt{Gata.1} interact atagonistically. Figure taken from \citealp{Graf09}.}\label{fig:01}
\end{figure}

\section{Approach}

For the single cell dynamic analysis to be possible, we were interested in gaining insight in single cell transcription data of both \texttt{Pu.1} and \texttt{Gata.1}. We combine two techniques for this to be possible: single cell tracking and protein tagging.

\subsection{Experimental Setting}

The data for the experiment were taken from Hoppe {\it et~al}, 2016 \citep{Hoppe16}. The expressed protein of interests \texttt{Pu.1} and \texttt{Gata.1} were tagged with distinct fluorescence proteins \texttt{eYFP} and \texttt{mCherry} respectively. Time-resolved expression levels were then measured based on the intensity of aforementioned florescence proteins.

Time-lapse imaging was performed at 37 $^{\circ}$C in fibronectin coated channel slides provided by Takara Bio. The light source for the experiment was a HXP 120 from Zeiss. 46HE and 43HE, both provided by Zeiss, were used to detect eYFP and mCherry fluorescence coming from the cells at exposure times between 400-1,500 ms using an AxioCam HRm from Zeiss. Bright field pictures were taken every 60-120 s while fluorescent pictures used for the measuring \texttt{PU.1-eYFP} and \texttt{GATA1-mCherry} were acquired every 30 min. Pictures used for analysis were saved in lossless TIF or PNG format. Single-cell tracking and image quantification were performed using self-written software as described by \citealp{Hoppe16}.

Figure \ref{fig:02} shows how the dataset from similarly conducted experiment taken from \citealp{Feigelman16}

\begin{figure}[h]
\includegraphics[width=8cm]{figures/cell-generations.png}
\caption{Example of single cell expression data. In this example, initially only one cell is observed. The cell would then divide into two cells. The system automatically recognizes this and separate measurements of expression would then be conducted, which is visible in the lower part of the figure. The above example also emphasizes the uniqueness and the tracability of each cell by pointing two specific lineages within the data: the blue and red lineages. The system records complete expression measurement of the lineages which could be seen in second half of the figure corresponding with the first half of the figure. The system is capable of recording not only cell division but also cell deaths, as can be seen through the x's sign in the upper part of the figure. Figure taken from \citealp{Feigelman16}.}\label{fig:02}
\end{figure}

\section{Models}

We utilize particle filtering to learn the dynamics of maturation of Common Myeloid Progentior (CMP) cells. The definition of particle filtering could be seen in the subsection \textbf{Particle Filtering}. \textbf{Simulation Process} details how the simulation is done and the probabilistic approaches is used in it. The assumed model of the cell maturation is then described in the subsection \textbf{Reaction Models}. A framework for model comparison is also presented in subsection \textbf{Models Comparison}. This is especially helpful for comparing the default model described in \textbf{Reaction Models} and other possible alternative models.

\subsection{Particle Filtering}

Particle filtering, which is also known as \textbf{Sequential Monte Carlo} (SMC) \citep{Doucet01, Liu98}), is a class of methods generally used to solve the filtering problem. Classically, filtering problem is the problem of identifying best estimate of some system given noisy and incomplete input -- hence the word filter \citep{DelMoral96}. As the alternative name implies, it is a set of methods that run Monte Carlo sequentially interrupted by inference phase. It consists mainly of two phase steps, \textbf{prediction-updating} and \textbf{mutation-selection}, that are done sequentially and repetitively. In the first phase, the prediction-updating updates the conditional probability of underlying assumption -- generally referred to as \textbf{posterior distribution} -- of a system given the previous simulation. Later on, a simulation (mutation) is done on given underlying assumption and simulations best fit the data would then be selected to update the underlying assumption in the next prediction-updating step \citep{DelMoral12}.

As the name also implies, particle filtering consists of sequentially and repetitively filtering of mathematical construct called particle, an abstract representation of combination of model, data and parameters. In the next part we will dive deep into the theory of particle filtering. First, we will define what a particle is. This is then followed by definition of Propensity, Posterior, Parameters and Prior. Finally the section will be closed by short review of other methods that also tried to stochastically infer parameters on time-lapsed fluorescent single cell data.

\subsubsection{Particle}

A particle $\mathcal{K}$ is defined as a triple of trajectory $X$, parameter set $\theta$ and assumed model $\mathcal{M}$,

\begin{equation}
\mathcal{K} := (X, \theta, \mathcal{M})\label{eq:01}
\end{equation}

Specifically in our case, trajectory refers to the concentration of measured proteins \texttt{Pu.1} and \texttt{GATA1} while parameter set encompasses all variables that were involved in the reaction such as propensity and its corresponding auxiliary variables (see subsection \textbf{Reaction Models}). Note that a trajectory is different from the experimental data $\mathcal{D}$. A trajectory is the result of the simulation done by applying the parameters onto our model. In functional notation we would assume the i-th value of the trajectory $X_i$ to be a function of the i-th value of the trajectory, the model and the parameters,

\begin{equation}
X_i = f(X_{i - 1}, \mathcal{M}, \theta_{i - 1})\label{eq:02}
\end{equation}

\subsubsection{Propensity}

As the name suggests, propensity is measure of how likely a certain thing would be happening in a given time. In probabilistic theory, it is understood as a tendency a certain type of physical situation would yield an outcome of certain kind \citep{Nodelman03}. In our model specifically, a propensity roughly corresponds to the chance of a reaction to happen in a given time,

\begin{equation}
a_i \propto P(R_k = r_i)
\end{equation}

Where $P(R_k = r_i)$ refers to the probability of reaction at time point k to be the -th reaction. For reactions $r_1, r_2, \cdots r_p$ there are propensities $a_1, a_2, \cdots a_p$ that are uniquely assigned to each reaction. The probability of a reaction $r_i$ to happen in a given time is thus the ratio between its propensity $a_i$ to the total sum of all propensities \citep{Gillespie77},

\begin{equation}
P(R_k = r_i) = \frac{a_i}{\sum_{j=0}^{p} a_i}
\end{equation}

In our model, two kinds of propensities were defined. For uninhibited reaction, i.e. a reaction in which no inhibiting effect from other species -- a term denoting any entity that is involved in a reaction network -- is assumed, the reaction specific propensity is defined as the parameter in the mass action law of chemical reaction:

\begin{equation}
a_i = k_i \label{eq:23}
\end{equation}

with $k_i$ referring to the reaction parameter of $r_i$.

For reaction $r_i$ inhibited by a species $Y$, Michaelis-Menten inhibition kinetics is assumed to be the propensity of the reaction \citep{MichaelisMenten13},

\begin{equation}
a_i = k_i \cdot (1 - \frac{X_i^{n}}{X_i^{n} + K_{Y}^{n}})\label{eq:24}
\end{equation}

where $k_i$, $X_i$ and $K_{Y}$ refer to reaction parameter of $r_i$, concentration of product $X$ in a given time and inhibiting coefficient of regulator $Y$ on $X$ respectively.

\subsubsection{Posterior}

Our posterior describes the probability of having the trajectory $X$ and parameter $\theta$ given the observation $\mathcal{D}$,

\begin{equation}
P(X, \theta | \mathcal{D})\label{eq:03}
\end{equation}

This is understood as the probability of having our simulation return a given set of values \textit{and} having the parameter set $\theta$ given that we previously observed the experimental data $\mathcal{D}$. Using Bayes' Theorem we could further expand our posterior into an update rule,

\begin{equation}
P(X, \theta | \mathcal{D}) = \frac{P(\mathcal{D} | X, \theta)  P(X, \theta)}{P(\mathcal{D})}\label{eq:04}
\end{equation}

In the simulation it is well known that, to compute prior $P(\mathcal{D} | X, \theta)$, only knowledge about the trajectory of the simulation is needed \citep{Feigelman16}. Hence, the prior could be simplified as,

\begin{equation}
P(\mathcal{D} | X, \theta) = P(\mathcal{D} | X)\label{eq:05}
\end{equation}

Note that the above equation inherently assumes that $\mathcal{D}$ is only directly dependent on $X$ and $X$ is in turn only directly dependent on $\theta$. Incorporating this onto our update rule, and expanding Equation (\ref{eq:04}) using chain rule, we get

\begin{equation}
P(X, \theta | \mathcal{D}) = \frac{P(\mathcal{D} | X)  P(X | \theta) \pi(\theta)}{P(\mathcal{D})}\label{eq:06}
\end{equation}

One of the interesting aspects of our method is the fact that the i-th simulation result is only dependent on previous simulation, a property known as \textit{Markov property}. We could thus rewrite the update rule as follow,

% TODO correct the second product definition. It seems off

%\begin{equation}
%P(X, \theta | \mathcal{D}) = \frac{\prod_{i=0}^{N} P(\mathcal{D}_i | X_i) P(X_0) \prod_{l=1}^{N} P(X_{[t-1, ti]} | X_i, \theta)\pi(\theta)}{P(\mathcal{D})}\label{eq:07}
%\end{equation}

\begin{equation}
P(X, \theta | \mathcal{D}) = \frac{\prod_{i=0}^{N} P(\mathcal{D}_i | X_i) P(X_0) \prod_{l=1}^{N} P(X_{[tl-1, tl]} | X_l, \theta)\pi(\theta)}{P(\mathcal{D})}\label{eq:07}
\end{equation}

\subsubsection{Parameters}

During the simulation we assumes certain parameters that would influence the trajectory of the simulation. The parameter set $\theta$ is a P-tuple containing all the parameters that are assumed in the simulation,

\begin{equation}
P := (P_0, P_1, \dots , P_P)\label{eq:08}
\end{equation}

\subsubsection{Prior}

Our prior $\pi(\theta)$ could be expanded by assuming the independence of each parameter $\theta_i$ within the parameter set $\theta$,

\begin{equation}
\pi(\theta) = \prod_{i=1}^{P} \pi(\theta_i)\label{eq:09}
\end{equation}

Specifically for this simulation, we assume our prior to be gamma distributed with the parameters $\alpha$ and $\beta$,

\begin{equation}
\pi(\theta) = \prod_{i=1}^{P} Ga(\theta_i, \alpha_i, \beta_i)\label{eq:10}
\end{equation}

We used gamma distribution as our prior in our model due the fact that the, according to \citep{Feigelman16}, the posterior of parameters would be in turn be gamma distributed, a fact which will be shown later in results section. It is thus computationally convenient to estimate the posterior if gamma prior is assumed. This phenomenon in which prior and posterior having the same distribution family is known in Bayesian Statistics as \textit{conjugate prior} \citep{Gelman14}.

\subsubsection{Comparison with other methods}

There are several methods that infer stochastic processes. Wilkinson -- and later on, Golightly and Wilkinson --  inferred stochastic differential equation (SDE) kinetic model of bacterial gene regulation using likelihood-free Markov Chain Monte Carlo (MCMC) on florescence data \citep{Wilkinson10, GolightlyWilkinson10}. Gonzales \textit{et al} did two stochastic inference methods, Mixed Effects (ME) and the Chemical Master Equation (CME), on time-lapsed single-cell data. The paper also tested the method by inferring previously known HOG pathway in yeast \textit{Saccharomyces cerevisiae} \citep{Gonzales13}. Koromowski \textit{et al} developed Bayesian framework performed using Markov Chain Monte Carlo (MCMC) and linear noise approximation to calibrate models on fluorescent experiment data \citep{Komorowski09}.


\subsection{Inference Process}

The simulation is run in the following main steps:

\begin{enumerate}
\item Initialization of parameters $\theta$.
\item Input of data $\mathcal{D}$.
\item Particle filtering routine:

\begin{enumerate}
\item Generation of initial particles for step i

\begin{equation}
Ki := (K_{i1}, K_{i2}, \dots, K_{im})\label{eq:11}
\end{equation}

\item Simulation run of each particle $K_{ij}$
\item Weighting of each particle. The weight is a function of the probability of observing the data given the simulation result.

\begin{equation}
w_i^k = P(D_i | X_i^k) = \mathcal{N}(\mathcal{D}_i | X_i^k)\label{eq:12}
\end{equation}

\item Parameter update for every K,

\begin{equation}
\theta^k \propto P(\theta | X^k_{[to, ti]})\label{eq:13}
\end{equation}

\end{enumerate}

\item Model comparison.

\end{enumerate}

The visualization of the whole particle filtering process could be seen in Figure \ref{fig:03}.

\begin{figure*}[h]
%\makebox[\textwidth]{\includegraphics[width=\paperwidth]{figures/particle_filtering.png}}
\includegraphics[width=15cm]{figures/particle_filtering_2.png}
\caption{Visualization of particle fil	tering. (A): The particle filter requires a series of observation $\mathcal{D} = (\mathcal{D}_0, \mathcal{D}_1, \cdots \mathcal{D}_N)$ (top) and a prior distribution for model parameters $\pi(\theta)$ (bottom) as input. (B): Particles are initialized by sampling initial states $X(K)$ for each particle $K$. Parameters $\Theta$ are sampled from the prior distribution $\pi(\theta)$. (C): The trajectories are resampled according to the likelihood $w_0^{(k)} = P(D_0|X_0(k))$ and propagated to the next time step using stochastic simulations to generate new states $X_1^{(k)}$ at timepoint $t_1$. Model parameters for each latent trajectory are resampled from the conditional distribution $P(\theta|X^{(k)}_{[t0,t1]})$. (D): At each iteration, the weights are recomputed and the particles are resampled. Resampled particles are propagated to the next timepoint and the parameters are resampled conditional on the resampled latent trajectories. Over time the posterior parameter distribution converges to the true value. Figure taken from Feigelman, 2016 \citealp{Feigelman16}}  \label{fig:03}
\end{figure*}

\vspace*{-6pt}

\subsubsection{Particle Generation}

As mentioned before, A particle $\mathcal{K}$ is defined as a 3-tuple of trajectory $X$, parameter set $\theta$ and model $\mathcal{M}$. For a specific model, j-th particle at i-th time particle is then just a tuple of trajectory and parameter set $k_{ij} = (x_i, \theta_{ij})$.

For initial time, a particle could be constructed by combining initial parameters, which were arbitrarily defined by a pre-defined prior, and initial trajectory value $X_0$. There are three ways to define the initial trajectory value. First is to take 0 as initial value. Second is to take the first measurement data $\mathcal{D}_0$. Third is to model initial data as normal distributed instances around $\mathcal{D}_0$, i.e. $X_0 \propto \mathcal{N}(\mathcal{D}_0)$. It is worth noting that the first strategy may not always be applicable.

M particles would then be created and for each iteration of the simulation with each particle having weight which calculated from previous iteration $w_{i - 1}^k$. In the next iteration, M particles are to be chosen. Note that this implies that a particle from previous iteration could be chosen more than once. Corollarly, a particle from previous iteration may also not be chosen at all. The draw is done by first drawing a random $z \propto \mathcal{U}(0, 1)$. A particle $k_{ij}$ is chosen with j denoting the smallest index so that the sum of all weight of particles with index smaller than j is larger than z, i.e.

\begin{equation}
\sum_{l=1}^{j} w_{i - 1}^l \geq z\label{eq:14}
\end{equation}

\subsubsection{Simulation Run}

Simulation is done using Gillespie's SSA algorithm \citep{Gillespie77}. Dating back to as early as 1945, when it was developed by Joseph Doob \citealp{Doob45} \citealp{Chung67}, modern Gillespie implementation includes improvements developed years after the invention of the algorithm such as $\tau$-leaping and Bayesian Approximation Method. Such improvements are essential since they would reduce computational cost of the simulation and in turn enable the simulation to scale further to accommodate more complex simulation involving larger systems -- in our case we could simulate more cells at the same time in possible more fine-grained time-resolved manner. In our case especially, such improvements are needed to accommodate the explosion of the number of cells within our culture. Specifically, an approximative method would prevent a bottleneck in simulation caused by the exponential increase of the number of living cells that have to be simulated.\\

\textbf{Tau Leaping}

Generally, tau-leaping works by performing all reactions happening for an interval of length tau before updating the parameters and propensity function \citep{Cao06}. By introducing this kind of approximation we allow more efficient simulation and thus increase the capability of the simulation to cope with larger systems. In our case, we use following tau,

\begin{equation}
\tau = \frac{1}{a_0} ln \Gamma\label{eq:15}
\end{equation}

with $a_0$ referring to total sum of reaction propensities and $\Gamma$ being uniformly distributed between 0 and 1. 

\subsubsection{Particle Weighting}

Upon the completion of all simulations within one iteration, the quality of each simulation (and in turn, the particle) will be quantified. This quantification is implied in the weighting $w_{i}^k$ of the particle at the next iteration. We assume the Gaussian distribution of particle around the measurement time at time $t=i$. Using this assumption, we can quantify our weight in following manner,

\begin{equation}
w_i^k = P(\mathcal{D}_i | X_i^k) = \mathcal{N}(\mathcal{D}_i | X_i^k)\label{eq:16}
\end{equation}

%To generate particle for next iteration we utilize, however, the normalized particle weight,

%$$w_j^k = \frac{w_j^k}{\sum_{l=1}^{M} w_{j}^l}$$

\subsubsection{Parameters Update and Gamma Distribution}

After each run, the parameters would then be updated using the underlying Gamma Distribution. For $k$-th particle, the updated parameters at time $i$ are dependent on previous parameters given the trajectory of data in $[t_0, t_i]$,\\

\begin{equation}
\Theta^k \propto P(\Theta | X_{[t_0, t_i]})\label{eq:17}
\end{equation}

Applying Bayes' Rule on the probability and independent assumption, we get following equation,

\begin{equation}
P(\Theta | X) = \frac{P(X | \Theta)}{P (X)} \prod_{i}^{P} P(\Theta_i)\label{eq:18}
\end{equation}

The Gamma Distribution is particularly favored for the update since a conjugate prior of Gamma Distribution is in turn Gamma distributed,

\begin{equation}
P(\Theta | X) = \frac{P(X | \Theta)}{P (X)} \prod_{d=1}^{P} Ga(\Theta_d, \alpha_d, \beta_d)\label{eq:19}
\end{equation}

With $\alpha$ and $\beta$ referring to both the shape and rate of the distribution respectively. Owing to the fact that conjugate prior of Gamma Distribution is also Gamma distributed, this could be then further simplified as an update rule of the Gamma distribution,

\begin{equation}
P(\Theta | X) = \frac{P(X | \Theta)}{P (X)} \prod_{d=1}^{P} Ga(\Theta_d, \alpha_d + r_d, \beta_d + Gd)\label{eq:20}
\end{equation}

$r_d$ here refers to the number of times the d-th reaction was fired during the run and $G_d$ is defined as follows,

\begin{equation}
G_d := \frac{1}{k_d} \sum_{l=0}^{i} a_{l}(X(S))\label{eq:21}
\end{equation}

Here, $k_d$ and $a_{l}(X(S))$ refer to reaction rate of the d-th reaction and trajectory $X(S)$-dependent propensity function, respectively.

\subsection{Reaction Models}

We assume following putative cross inhibition between genes \texttt{Pu.1} and \texttt{Gata.1} playing roles in cell differentiation dynamics:

\begin{center}
\begin{tikzpicture}[node distance=2cm]

\node (N0) {\Large $\varnothing$};
\node [right of = N0] (N1) {\Large $\varnothing$};
\node [below of = N0] (R0) {\Large \texttt{Pu.1}};
\node [below of = N1] (R1) {\Large \texttt{GATA1}};
\node [below of = R0] (N2) {\Large $\varnothing$};
\node [below of = R1] (N3) {\Large $\varnothing$};

\draw [->,ultra thick,blue] (N0) -- node[left, black] {$k_1$} (R0);
\draw [->,ultra thick,blue] (N1) -- node[right, black] {$k_2$} (R1);
\draw [->,ultra thick,blue] (R0) -- node[left, black] {$k_3$} (N2);
\draw [->,ultra thick,blue] (R1) -- node[right, black] {$k_4$} (N3);

\draw [-|,ultra thick,blue]
($ (R0) + (3mm,-2.75mm) $)
-- +(0,-3mm) %node[below, black] {$v_3$}
-| ($ (R1) - (3mm,3mm) $) ;

\draw [-|,ultra thick,blue]
($ (R1) + (-3mm,2.75mm) $)
-- +(0,3mm) %node[above, black] {$v_3$}
-| ($ (R0) - (-3mm,-3mm) $);

\end{tikzpicture}
\end{center}

There are four possible reactions happening within our model: $k_1$, $k_2$, $k_3$ and $k_4$. In our simulation, each reaction was assigned propensity value which roughly is proportional to the likelihood of the reaction happening in a given time:

\begin{equation}
P(R_i = k_l) \propto a_l\label{eq:22}
\end{equation}

The propensity $a$ of decay reactions $k_3$ and $k_4$ roughly follows the mass action law of chemical reaction:

\begin{equation}
a_l = k_l \cdot A {;} \; l \in [3, 4]\label{eq:23}
\end{equation}

with $A_3$ and $A_4$ referring to number of \texttt{Pu.1} and \texttt{GATA1} respectively. For production reactions $k_1$ and $k_2$ we assume Michaelis-Menten inhibition kinetics that influences the production of the species. In this assumption, the inhibiting characteristics of a species on another species negatively influences the creation rate of the species it inhibits. The propensity of the reactions are thus,

\begin{equation}
a_1 = k_1 \cdot (1 - \frac{\texttt{GATA1}^{n}}{\texttt{GATA1}^{n} + K_{P1}^{n}})\label{eq:24}
\end{equation}

\begin{equation}
a_2 = k_2 \cdot (1 - \frac{\texttt{Pu.1}^{n}}{\texttt{Pu.1}^{n} + K_{P2}^{n}})\label{eq:25}
\end{equation}

With $K_{P1}$ and $K_{P2}$ referring to the respective protein levels at which inhibiton trough \texttt{GATA1} or \texttt{Pu.1} is at half its maximum effect.

\subsection{Models Comparison}

Our framework enables us to test not only standard hematopoietic stem cell maturation as described above, it also allows us to compare it different models. One method that could be used to compare models is Bayes Factors. It is done by performing the ratio of the posterior probabilities of two models $M_1$ and $M_2$,

\begin{equation}
B_{M1, M2} = \frac{P(M_1 | \mathcal{D})}{P(M_2 | \mathcal{D})}\label{eq:26}
\end{equation}

Using Bayes' Theorem, we can formulate the marginal probability as,

\begin{equation}
P(M | \mathcal{D}) = \frac{P(\mathcal{D} | M) P(M)}{P(\mathcal{D})}\label{eq:27}
\end{equation}

The marginal likelihood of model $P(\mathcal{D} | M)$ could be approximated using the particles at each iteration i (\citealp{Wilkinson11}). Since the observation of particles only depends on the previous particle, We could therefore rewrite the term as,

\begin{equation}
P(\mathcal{D} | M) = P(\mathcal{D_0}) \prod_{l=1}^{N} P(\mathcal{D}_{i+1} | \mathcal{D}_{0:i}, M)\label{eq:28}
\end{equation}

Assuming a priori equally likely models, the factor of $P(D)$ in (\ref{eq:26}) cancels between the two models and the Bayes Factors now becomes the ratio of two marginal likelihood,

\begin{equation}
B_{M1, M2} = \frac{P(\mathcal{D} | M_1)}{P(\mathcal{D} | M_2)}\label{eq:29}
\end{equation}

Besides Bayes Factors, there also other methods commonly used to compare methods such as Akaike Information Criterion and Bayesian Information Criterion \citep{PosadaBuckley14, Bozdogan87}

\section{Result}

In order to assess the performance of parameter inference and model comparison via particle filtering we created a synthetic data set of time series measurements	. The reaction model decribed in 3.3 was used as basis to generate the set with the Gillespie algorithm (Gillespie, 1977). As the model describes a bistable switch the test data needs to contain sufficient information regarding both dominance states as well as transition phases between the states. Dominance states are defined as a time frame in which one transcription factor level rises to degree that noticeably inhibits the production rate of the other transcription factor. The simulation starts with a single cell, which would then divide into two daugthers cell. These cells inherit the protein levels of their parent and continue their simulation independently. With enough cell divisions the resulting time series tree can accurately represent the spectrum of possible system configurations. For our data set we choose 5 cell divisions resulting in 32 trajectories at the end of simulation. After generation of the data the particle filter was applied with a particle count of 1000 and replicated two times to investigate the consistency of the parameter inference.

\begin{table}[h]
\begin{center}
\begin{tabular}{l | c | c | c | c | c }
Parameter & Simulation & Prior & Replicate 1 & Replicate 2 & Replicate 3 \\
\hline
\texttt{$k_1$} [1/s]& 120 & 100 & 104.489 & 105.533 & 104.497\\
\texttt{$k_2$} [1/s]& 120 & 100 & 104.585 & 104.771 & 104.099\\
\texttt{$k_3$} [1/s]& 0.4 & 0.3 & 0.356 & 0.351 & 0.344\\
\texttt{$k_4$} [1/s]& 0.4 & 0.3 & 0.355 & 0.337 & 0.356\\
\texttt{$K_{P1}$} & 170 & 190 & 189.141 & 189.114 & 189.13\\
\texttt{$K_{P2}$} & 170 & 190 & 188.209 & 189.295 & 189.606\\
\end{tabular}

\vspace*{7pt}
\caption{Exact value of the parameters used in the simulation to generate the test data and expected value of the prior used for particle filtering. Replicates show the expected value of the posterior distributions for each particle filtering run on the test data set.}
\label{tab:01}
\end{center}
\end{table}

\begin{figure*}[h]
\includegraphics[width=\textwidth]{figures/test_results.pdf}
\caption{Result of the particle filtering algorithm for each parameter of the reaction model on the test data set. The dashed lines indictated the optimal parameter values used in the data creation. The red lines show the prior distribution for the parameters used in the particle filtering. Replicate lines represent the posterior distribution of fitted paramters after particle filter run.}  \label{fig:04}
\end{figure*}

Table \ref{tab:01} shows the results of the particle filtering test for each of the 6 paramters with the posterior distributions visualized in figure \ref{fig:04}. The posterior distribution of the paramters $k_3$ and $k_4$ (the degradation reactions of \texttt{Pu.1} and \texttt{Gata1} respectivly) show the most recognisable motion towards their optimal values while the parameters $k_1$ and $k_2$ responsible for the production reactions show only little movement. No significant movement of the posteriors representing the inhibition paramters $K_{P1}$ and $K_{P2}$ can be observed. The replicates of the particle filtering are highly consistent in their results with expected values of their posterior distributions being marginally different. Furthermore the paramters $k_3$ and $k_4$ show a similar transformation of their particluar priors towards the shape of the resulting posteriors.


In order to investigate the capability of the particle filter for model comparsion a second model was created from the reaction model described in section 3.3. For this model the inhibitory kinetics of the production reactions were removed resulting in complete independence of protein levels between \texttt{Pu.1} and \texttt{Gata1}. The production parameters $k_1$ and $k_2$ were lowered from 120 to 90 to compensate for the removal of the inhibitory effect. Afterwards the particle filter algorithm was applied to test data set with 1000 particles for both the original reaction model with optimal paramters and the modified model without cross inhibiton. By tracking the particle weight of a particle trajectory at each given datapoint the particle filter can infer the likelihood for the Bayes Factors described in equation \ref{eq:29}. For ease of computation the weights were transformed to log probabilities with base 10, resulting in the following Bayes Factor:


\begin{equation}
B_{M1, M2} = \frac{P(\mathcal{D} | M_{inhibitory})}{P(\mathcal{D} | M_{indepent})} = \frac{10^{-2221.498}}{10^{-2234.828}} \approx 2.14 \cdot 10^{13} \label{eq:30}
\end{equation}

The inferred Bayes Factor shows a significantly increased likelihood of the inhibitory model used in the generation of test data set, compared to the indepent model, consequently identifying the correct model.
\\
\\ % prior choiche ? tree picture
After testing the particle filter was then applied to the time series data set containing the cell lineage of a murine stem cells. 3 trees were chosen based on their protein distribution showing states of dominance for both transcription factor levels. With respect to the toggle switch hypothesis these trees sufficiently represent the spectrum of lineage choice possibilities. The particle filter algortihm was performed with 1000 particles for all 3 trees, each containing respectively 41, 25 and 42 living cells at the end of the experiment.

\begin{table}[h]
\begin{center}
\begin{tabular}{l | c | c | c | c }
Parameter & Prior & Tree 1 & Tree 2 & Tree 3 \\
\hline
\texttt{$k_1$} [1/s]& 800 & 799.369 & 798.596 & 800.094\\
\texttt{$k_2$} [1/s]& 400 & 398.35 & 401.315 & 400.126\\
\texttt{$k_3$} [1/s]& 0.01 & 0.0129 & 0.0125 & 0.0128\\
\texttt{$k_4$} [1/s]& 0.01 &  0.0185 & 0.0159 & 0.0223\\
\texttt{$K_{P1}$} & 4000 & 4002.256 & 3993.389 & 4001.58\\
\texttt{$K_{P2}$} & 15000 & 15002.5 & 14998.55 & 14993.86\\
\end{tabular}

\vspace*{7pt}
\caption{Expected value of the prior used for particle filtering for each parameter. Trees show the expected value of the posterior distributions for each particle filtering run on the respective cell lineage tree.}
\label{tab:02}
\end{center}
\end{table}

Table \ref{tab:02} shows the fitted paramters derived from using particle filtering on the time series data of murine stem cells with the corresponding prior and posterior distribution shown in figure \ref{fig:05}. The particle filter exhibits a similar paramter fitting performance compared to the test data set. On both the simulated as well as the experimental data the degradation rate parameters $k_3$ and $k_4$ show signifcant movement away from their expected prior values while the inhibitation paramters $K_{P1}$ and $K_{P1}$ show no change of their expected value. However the production parameters $k_1$ and $k_2$ on experimential data set were not changed by the particle filtering.

After perfoming the particle filtering with the reaction model, the independent model used for model comparison on the test data set, was applied to experimental data an the following likelyhoods were derived:

\begin{equation}
B_{M1, M2} = \frac{P(\mathcal{D} | M_{inhibitory})}{P(\mathcal{D} | M_{indepent})} = \frac{10^{-884.2323}}{10^{-1096.024}} \approx 6.19 \cdot 10^{211} \label{eq:31}
\end{equation}

As shown by the Bayes Factor the inhbitory model massively outperforms the independent model on the experimental data set.


\begin{figure*}[h]
\includegraphics[width=\textwidth]{figures/real_data.pdf}
\caption{Result of the particle filtering algorithm for each parameter of the reaction model on timer series data of murine stem cells. The red lines show the prior distribution for the parameters used in the particle filtering. Tree lines represent the posterior distribution of fitted paramters after particle filter run  on the respective cell lineage tree.}  \label{fig:05}
\end{figure*}

\section{Discussion}

The test of the particle filtering algorithm on the simulated demonstrates the frameworks capability to infer model paramters based on tree structured time series data. However it shows uneven fitting performance given different paramter enviroments, with the degradation rates $k_3$ and $k_4$ being moved signficantly towards their optimum, while the production rate paramters $k_1$ and $k_2$ exhibit slower movement and the inhibitation parameters $K_{P1}$ and $K_{P1}$ were not changed at all. We suspect that this disparity of fitting performance is due to the difference in impact of the individual paramters on the protein levels and therefore the weight of the particles. The degredation reaction propensities depend on both the protein levels as well as the degredation rate and thus small changes of rates can have signifcant influence on the protein levels. In comparison the production reaction propensities do not get scaled with protein numbers and hence the particle weighting is less susceptible to changes of production rates, resulting in a reduced fitting performance. As the inhibitation parameter only negativly scale the production reaction propensites, changes of these parameters shows even less impact on protein levels and the resulting particle weights. Futhermore inhibitation parameters represent absolute protein numbers that influence rates rather than being actual reaction rates, potentially hindering the effectiveness of particle filter gamma update. Another factor influencing the performance of the particle filter paramter inference is presence of equilibrium states in time series, during which the protein levels do not change by large margin. In these time frames the production and the degradation reaction propensities need to be evenly matched. This however can be achieved with any suboptimal production rate as long as the matching degredation rate can keep the equilibrium. In reaction model used during testing both production and degradtion rates lie below the optimal values, possibly resulting in paramter systm that can keep equilibrium stable. While the test data set also contains a multitude of transition phases with high alteration of protein levels, these phases are potentially not enough to allow for fitting of the parameters through particle filtering. This assumption is further amplified by remarkable consistency of the replicated particle filter applications on the test data.


The model comparison of the inhbitory and the indipendant model shows promising results as the Bayes Factor can accuratly discern which model was used for generate of the test data set by a large margin. This shows that although the independent model has the possibility to generate trajectories similiar to that of an inhibitory model, the resulting performance of the particle filtering is still negatively influenced.



The interpretation of the particle filtering on the experimental time series trees proves to be ambiguous. While the particle filter managed to significantly alter the degradation rates $k_3$ and $k_4$ with $k_4$ being more than doubled on Tree 3, the other parameters remain unchanged at the expected value of their respective priors. While this behavior is similar to the particle filter performance of the reaction model on the test data set, the results are insuffienct to derive whether	the underlying mechanics can be represented through the model. As all particle filtering instances on the experimental data show to be highly consistent, the performance appears to be unaffected by potential disparity of the experimental data. One possible factor influencing the performance of the particle is the number of transition phases in the experimental data set. If the initial assumption of toggle switch deciding the cell fate holds true, then most of cells would spend large quantity of the experiment time in moderate equilibrium state until one transcription factor reaches a dominance threshold by chance, triggering a transition phase. As this transition represent the lineage choice, no futher transition phases can be observed in this branch of the cell lineage tree and protein levels are then governed by the underlying mechanism of MEP or GMP cells. This change of mechanical ruling could potentially impair the particle filters ability to infer parameters for the original reaction model.


Another possible explanation for particle filter performance is a discrepance between the model used for simulation and the underlying mechanics govering the transcription factors. The utilized reaction model is rather coarse as only contain production and degradation reactions. In comparison a more fine grained model containing the state of transcription sites corresponding to the proteins as well as mRNA levels, production, degradation and translation ratios could potentially better represent the stochastic process of the fundamental mechanics. Regarding the model representing the intial hypothesis of a cross inhibiton we mindful of a recent study on same experimental data set claiming independence of early myeloid lineage choice with regards to the transcription factors \texttt{Pu.1} and \texttt{Gata.1} (\citealp{Hoppe16}). In their work the authors make sound arguements against a lineage choice governed by a stochastic process between \texttt{Pu.1} and \texttt{Gata.1}. If the authors suspicions holds true, then the rise of transcription factor production would be regulated by an external factor, majorly influencing the particle filters ability to infer production rate parameters during transition phases. However the model comparison between the indepent and the inhibitory model on the experimental data show a massive favor towards the inhibitory model, suggesting that cross talk between the transcription factors is definitively present. This is in accordance with the observation that at no time in the experiment the highest levels of both transcription could be observed simultaneously. Ultimately the described finding suggest the role of a cross inhibitory model to be the one of executing the lineage choiche rather than making it.



% challenges

\section{Conclusion}

In the course of this project we managed to implement the particle filter algorithm as a framework for the inference of model parameters through the use of time series data. We demonstrate its ability to parameterize models with remarkable consistency and show the usefulness of particle trajectories for model comparison via Bayes Factors. While the particle filter did not manage to parameterize an agreeable model of our initial hypothesis, we found this to be in accordance with a recent study on the same data set, questining the role \textit{Gata1} and \textit{Pu.1} as toggle swtich deciding the cell fate of hematopoietic stem cells. This highlights our framework's capability to convey underlying mechanics prior unknown to us. This in turn hallmarks our framework potential as supporting tool for research scientists.

\vspace*{-6pt}

%\begin{figure}[!tpb]%figure1
%\fboxsep=0pt\colorbox{gray}{\begin{minipage}[t]{235pt} \vbox to 100pt{\vfill\hbox to
%235pt{\hfill\fontsize{24pt}{24pt}\selectfont FPO\hfill}\vfill}
%\end{minipage}}
%%\centerline{\includegraphics{fig01.eps}}
%\caption{Caption, caption.}\label{fig:01}
%\end{figure}

%\begin{figure}[!tpb]%figure2
%%\centerline{\includegraphics{fig02.eps}}
%\caption{Caption, caption.}\label{fig:02}
%\end{figure}


\section*{Acknowledgements}

We would like to express our gratitude to Werner Mewes, Fabian Theis, Jan Quell and Anna Dieckman for facilitating the project. We would also like to thanks Philipp S. Hoppe, Michael Schwarzfischer, Dirk Loeffler, Konstantinos D. Kokkaliaris, Oliver Hilsenbeck, Nadine Moritz, Max Endele, Adam Filipczyk, Adriana Gambardella, Nouraiz Ahmed, Martin Etzrodt, Daniel L. Coutu, Michael A. Rieger, Bernhard Schauberger, Ingo Burtscher, Olga Ermakova, Antje B\"urger, Heiko Lickert, Claus Nerlov and Timm Schroder who agreed to provide us with the experiment data, without which we wouldn't be able to test our framework and models. 

%\section*{Funding}
%We acknowledge the fact that the project and corresponding projects this project is based on are, directly and undirectly, financed by several grants given to following institutes within the Helmholtz Zentrum M\"unchen: the Institute of Bioinformatics and Systems Biology, the Institute of Computational Biology, the Institute of Diabetes and Regeneration Research, the Institute of Development Genetics and the Research Unit Stem Cell Dynamics. Some grants might also be involved coming from following institutions: Department of Biosystems Science and Engineering at ETH Zurich, MRC Molecular Haematology Unit at University of Oxford, Institute di Biologia Cellulare e Neurobiologia at CNR, Beta Cell Biology at the Medical Faculty of the Technical University of Munich, Department of Mathematics at the Technical University of Munich and EMBL Mouse Biology Unit.\vspace*{-12pt}

%\bibliographystyle{natbib}
%\bibliographystyle{achemnat}
%\bibliographystyle{plainnat}
%\bibliographystyle{abbrv}
%\bibliographystyle{bioinformatics}
%
%\bibliographystyle{plain}
%
%\bibliography{Document}

\vspace*{15pt}

\begin{thebibliography}{}

\bibitem[Feigelman, 2016]{Feigelman16}
Feigelman, J. (2016). "Stochastic and deterministic methods for the analysis of Nanog dynamics in mouse embryonic stem cells." PhD Thesis, Technische Universit\"at M\"unchen, Munich, Germany.\vspace*{-2pt}

\bibitem[Hoppe {\it et~al}, 2016]{Hoppe16}
Hoppe, P.S., Schwarzfischer, M., Loeffler, D.,  Kokkaliaris, K.D., Hilsenbeck, O., Mortz, N., ... \& Etzrodt, M. (2016). Early myeloid lineage choice is not initiated by random PU.1 to GATA1 protein ratios. \textit{Nature}, \textbf{535(7611)}, 299-302.

\bibitem[Graf \& Enver, 2009]{Graf09}
Graf, T., \& Enver, T. (2009). Forcing cells to change lineages. \textit{Nature}, \textbf{462(7273)}, 587-594.

\bibitem[Elowitz {\it et~al}, 2016]{Elowitz02}
Elowitz, M. B., Levine, A. J., Siggia, E. D., \& Swain, P. S. (2002). Stochastic gene expression in a single cell. \textit{Science}, \textbf{297(5584)}, 1183-1186.

\bibitem[Orkin {\it et~al}, 2008]{Orkin08}
Orkin, S. H., \& Zon, L. I. (2008). Hematopoiesis: an evolving paradigm for stem cell biology. \textit{Cell}, \textbf{132(4)}, 631-644.

\bibitem[Filipczyk {\it et~al}, 2015]{Filipczyk15}
Filipczyk, A., Marr, C., Hastreiter, S., Feigelman, J., Schwarzfischer, M., Hoppe, P. S., ... \& Hilsenbeck, O. (2015). Network plasticity of pluripotency transcription factors in embryonic stem cells. \textit{Nature cell biology}.

\bibitem[Zhang {\it et~al}, 2003]{Zhang03}
Zhang, J., Niu, C., Ye, L., Huang, H., He, X., Tong, W. G., ... \& Harris, S. (2003). Identification of the haematopoietic stem cell niche and control of the niche size. \textit{Nature}, \textbf{425(6960)}, 836-841.

\bibitem[Doucet {\it et~al}, 1945]{Doucet01}
Doucet, A., De Freitas, N., \& Gordon, N. (2001). An introduction to sequential Monte Carlo methods. In \textit{Sequential Monte Carlo methods in practice} (pp. 3-14). Springer New York.

\bibitem[Liu and Chen, 1998]{Liu98}
Liu, J. S., \& Chen, R. (1998). Sequential Monte Carlo methods for dynamic systems. \textit{Journal of the American statistical association}, 93(443), 1032-1044.

\bibitem[Del Moral, 1996]{DelMoral96}
Del Moral, P. (1996). Non-linear filtering: interacting particle resolution. \textit{Markov processes and related fields}, \textbf{2(4)}, 555-581.

\bibitem[Del Moral, 2012]{DelMoral12}
Del Moral, P., Doucet, A., \& Jasra, A. (2012). On adaptive resampling strategies for sequential Monte Carlo methods. \textit{Bernoulli}, \textbf{18(1)}, 252-278.

\bibitem[Gillespie, 1977]{Gillespie77}
Gillespie, D. T. (1977). Exact stochastic simulation of coupled chemical reactions. \textit{The journal of physical chemistry}, \textbf{81(25)}, 2340-2361.

\bibitem[Doob, 1945]{Doob45}
Doob, J. L. (1945). Markoff chains--denumerable case. \textit{Transactions of the American Mathematical Society}, \textbf{58(3)}, 455-473.

\bibitem[Chung, 1967]{Chung67}
Chung, K. L. (1967). \textit{Markov Chain}. Berlin: Springer-Verlag.

\bibitem[Gillespie, 2007]{Gillespie07}
Gillespie, D. T. (2007). Stochastic simulation of chemical kinetics. \textit{Annu. Rev. Phys. Chem.}, \textbf{58}, 35-55.

\bibitem[Fahrmeir {\it et~al}., 2007]{Fahrmeir07}
Fahrmeir, L., K\"unstler, R., Pigeot, I., \& Tutz, G. (2007). \textit{Statistik: Der Weg zur Datenanalyse}. Springer-Verlag.

\bibitem[Wilkinson, 2011]{Wilkinson11}
Wilkinson, D. J. (2011). Stochastic modelling for systems biology. CRC press.

\bibitem[Zechner {\it et~al}., 2011]{Zechner11}
Zechner, C., Pelet, S., Peter, M., \& Koeppl, H. (2011, December). Recursive Bayesian estimation of stochastic rate constants from heterogeneous cell populations. In \textit{2011 50th IEEE Conference on Decision and Control and European Control Conference} (pp. 5837-5843). IEEE.

\bibitem[Haseltine {\it et~al}., 2005]{Haseltine05}
Haseltine, E. L., \& Rawlings, J. B. (2005). On the origins of approximations for stochastic chemical kinetics. \textit{The Journal of chemical physics}, \textbf{123(16)}, 164115.

\bibitem[Sherlock {\it et~al}., 2014]{Sherlock14}
Sherlock, C., Golightly, A., \& Gillespie, C. S. (2014). Bayesian inference for hybrid discrete-continuous stochastic kinetic models. \textit{Inverse Problems}, \textbf{30(11)}, 114005.

\bibitem[Posada and Buckley, 2014]{PosadaBuckley14}
Posada, D., \& Buckley, T. R. (2004). Model selection and model averaging in phylogenetics: advantages of Akaike information criterion and Bayesian approaches over likelihood ratio tests. \textit{Systematic biology}, \textbf{53(5)}, 793-808.

\bibitem[Bozdogan, 1987]{Bozdogan87}
Bozdogan, H. (1987). Model selection and Akaike's information criterion (AIC): The general theory and its analytical extensions. \textit{Psychometrika}, \textbf{52(3)}, 345-370.

\bibitem[Wilkinson, 2010]{Wilkinson10}
Wilkinson, D. J. (2010). Parameter inference for stochastic kinetic models of bacterial gene regulation: a Bayesian approach to systems biology. In \textit{Proceedings of 9th Valencia International Meeting on Bayesian Statistics}. Oxford University Press (pp. 679-705).

\bibitem[Golightly and Wilkinson, 2010]{GolightlyWilkinson10}
Golightly, A., \& Wilkinson, D. J. (2010). Markov chain Monte Carlo algorithms for SDE parameter estimation. \textit{Learning and Inference for Computational Systems Biology}, 253-276.

\bibitem[Gonzales {\it et~al}., 2013]{Gonzales13}
Gonzalez, A., Uhlendorf, J., Schaul, J., Cinquemani, E., Batt, G., \& Ferrari-Trecate, G. (2013). \textit{Identification of biological models from single-cell data: a comparison between mixed-effects and moment-based inference} (Doctoral dissertation, INRIA).

\bibitem[Komorowski {\it et~al}., 2009]{Komorowski09}
Komorowski, M., Finkenst\"adt, B., Harper, C. V., \& Rand, D. A. (2009). Bayesian inference of biochemical kinetic parameters using the linear noise approximation. \textit{BMC bioinformatics}, \textbf{10(1)}, 1.

\bibitem[Michaelis \& Menten, 1913]{MichaelisMenten13}
Michaelis, L., \& Menten, M. L. (1913). Die kinetik der invertinwirkung. \textit{Biochem. z}, \textbf{49(333-369)}, 352.

\bibitem[Gelman {\it et~al}., 2009]{Gelman14}
Gelman, A., Carlin, J. B., Stern, H. S., \& Rubin, D. B. (2014). \textit{Bayesian data analysis} (Vol. 2). Boca Raton, FL, USA: Chapman \& Hall/CRC.

\bibitem[Nodelman {\it et~al}., 2003]{Nodelman03}
Nodelman, U., Allen, C., \& Perry, J. (2003). \textit{Stanford encyclopedia of philosophy}.

\bibitem[Cao {\it et~al}., 2003]{Cao06}
Cao, Y., Gillespie, D. T., \& Petzold, L. R. (2006). Efficient step size selection for the tau-leaping simulation method. \textit{The Journal of chemical physics}, \textbf{124(4)}, 044109.

\end{thebibliography}
\end{document}
